src/main.py

```diff
@@ -180,15 +180,19 @@ def _generate_or_load_offline_dataset(env: BaseEnv,
 
 
 def _generate_interaction_results(
-    env: BaseEnv, teacher: Teacher, requests: Sequence[InteractionRequest]
+        env: BaseEnv,
+        teacher: Teacher,
+        requests: Sequence[InteractionRequest],
+        cycle_num: Optional[int] = None
 ) -> Tuple[List[InteractionResult], float]:
     """Given a sequence of InteractionRequest objects, handle the requests and
     return a list of InteractionResult objects."""
     print("Generating interaction results...")
     results = []
     query_cost = 0.0
     for request in requests:
-        monitor = TeacherInteractionMonitor(request, teacher)
+        monitor = TeacherInteractionMonitorWithVideo(env.render, request,
```

Comment:
I know we're going around and around on this, sorry 😬 but... now that we have both `TeacherInteractionMonitorWithVideo` and `TeacherInteractionMonitor`, maybe we should go back to something like the `if` statement that you had originally, and create a `TeacherInteractionMonitorWithVideo` here if `CFG.make_interaction_videos` is True, and `TeacherInteractionMonitor` otherwise? if you make this change, don't forget to remove `CFG.make_interaction_videos` within `TeacherInteractionMonitorWithVideo`.

but if this makes testing annoying, feel free to ignore me (unless @ronuchit ) chimes in strongly

Reply:
sgtm, yeah, in that case `TeacherInteractionMonitorWithVideo` doesn't need to care about `CFG.make_interaction_videos`

Reply:
but that might make mypy typing confusing.... fine to leave it too

Reply:
ok yeah mypy is not happy... should i override stuff or leave it as is?

Reply:
All good let's leave it